common/djangoapps/student/forms.py

```diff
@@ -234,3 +235,51 @@ def cleaned_extended_profile(self):
             for key, value in self.cleaned_data.items()
             if key in self.extended_profile_fields and value is not None
         }
+
+
+class DummyRegistrationExtensionForm(forms.Form):
+    """
+    This form is a placeholder form that can be overwritten by changing the value of
+    settings.REGISTRATION_EXTENSION_FORM. This form should be a standard Django form
+    with a save() method which takes a user argument.
+
+    A normal Django Model form is not quite suitable here, as it likely to require a model
+    which has a foreign key to the user, but the user will not yet be saved, so validation
+    will not complete.
+    """
+
+    def save(self, user=None):
+        """
+        Dummy save method.
+        """
+        pass
+
+    class Meta(object):
+        """
+        The serialization_options dictionary is used to help fill in the gaps that Django's form
+        system does not provide in order to use the 'FormDescription.add_field' function in
+        core.djangoapps.user_api.helpers. The settings that can't automatically be derived are:
+
+        * default
+        * include_default_option
+        * field_type (but only if you're using a field not in RegistrationView.FIELD_TYPE_MAP)
+
+        All other options can be mapped. For instance 'instructions' is populated from the standard 'help_text'
+        argument on the field. See openedx.core.djangoapps.user_api.views.RegistrationView.get to see how field
+        arguments are mapped.
+        """
+        serialization_options = {}
+
+
+def get_custom_form(*args, **kwargs):
+    """
+    Convenience function for getting the custom form set in settings.REGISTRATION_EXTENSION_FORM.
+
+    If settings.FEATURES["ENABLE_COMBINED_LOGIN_REGISTRATION"] is False, the dummy form is used regardless
+    of the setting.
+    """
+    if not settings.FEATURES.get("ENABLE_COMBINED_LOGIN_REGISTRATION"):
+        return DummyRegistrationExtensionForm(*args, **kwargs)
```

Comment:
I'm ok with removing the example form, to be clear, if you'd just rather there be notes on what to create instead of this example code, and to have the function return None. The difficulty is in shipping a working example.